pkgs/tools/networking/python-eduvpn-client/default.nix

```diff
@@ -0,0 +1,47 @@
+{ pkgs, stdenv, fetchFromGitHub, gdk_pixbuf, gobjectIntrospection, gtk3
+, libnotify, networkmanager_openvpn, pythonPackages, wrapGAppsHook }:
+
+pythonPackages.buildPythonApplication rec {
+  pname = "eduvpn-client";
+  version = "1.0.1";
+
+  src = fetchFromGitHub {
+    owner = "eduvpn";
+    repo = "python-${pname}";
+    rev = version;
+    sha256 = "1gmcnqa3ab8a2m9snhr3iqqx8vfjl180cfqhvwhigwf4iivpc5lx";
+  };
+
+  prePatch = ''
+    substituteInPlace eduvpn/util.py \
+      --replace "/usr/local" "$out"
+  '';
+
+  checkInputs = with pythonPackages; [ pytest mock pytestrunner ];
+  nativeBuildInputs = with pythonPackages; [ wrapGAppsHook ];
+
+  propagatedBuildInputs = with pythonPackages; [
+    # Non-Python Dependencies
+    gdk_pixbuf gobjectIntrospection gtk3 libnotify networkmanager_openvpn
+    # Python Dependencies
+    configparser dateutil dbus-python future pygobject3 pynacl requests
+    requests_oauthlib six repoze_lru pillow qrcode pytestrunner
+    ];
+
+  # Checks need X environment
```

Comment:
Providing a headless X11 environment shouldn't be a problem. For me it seems to require networkmanager first:
```
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.NetworkManager was not provided by any .service files
```